src/statement-list/statement_list_parse_tree.cpp

```diff
@@ -11,99 +11,77 @@ Author: Matthias Weiss, matthias.weiss@diffblue.com
-  }
+void statement_list_parse_treet::tia_modulet::add_var_inout_entry(
+  var_declarationt &declaration)
+{
+  var_inout.push_back(declaration);
 }
 
-void statement_list_parse_treet::functiont::output(std::ostream &out) const
+void statement_list_parse_treet::tia_modulet::add_var_output_entry(
+  var_declarationt &declaration)
 {
-  out << "Name: " << name << "\n";
-  out << "Version: " << version << "\n\n";
+  var_output.push_back(declaration);
+}
 
-  out << "--------- Input Variables ----------\n\n";
-  for(const auto &declaration : var_input)
-  {
-    declaration.output(out);
-    out << "\n\n";
-  }
+void statement_list_parse_treet::tia_modulet::add_var_constant_entry(
+  var_declarationt &declaration)
+{
+  var_constant.push_back(declaration);
+}
 
-  out << "--------- In/Out Variables ---------\n\n";
-  for(const auto &declaration : var_inout)
-  {
-    declaration.output(out);
-    out << "\n\n";
-  }
+void statement_list_parse_treet::tia_modulet::add_var_temp_entry(
+  var_declarationt &declaration)
+{
+  var_temp.push_back(declaration);
+}
 
-  out << "--------- Output Variables ---------\n\n";
-  for(const auto &declaration : var_output)
-  {
-    declaration.output(out);
-    out << "\n\n";
-  }
+void statement_list_parse_treet::function_blockt::add_var_static_entry(
+  var_declarationt &declaration)
+{
+  var_static.push_back(declaration);
+}
+
+void statement_list_parse_treet::tia_modulet::add_network(networkt &network)
+{
+  networks.push_back(network);
+}
 
-  out << "------------- Networks -------------\n\n";
-  for(const auto &network : networks)
-  {
-    network.output(out);
-    out << "\n";
-  }
+void statement_list_parse_treet::networkt::set_title(std::string &value)
+{
+  if(value.empty())
+    title.reset();
+  else
+    title = value;
 }
 
-void statement_list_parse_treet::function_blockt::output(
-  std::ostream &out) const
+statement_list_parse_treet::networkt::networkt(std::string &title)
 {
-  statement_list_parse_treet::functiont::output(out);
+  set_title(title);
 }
 
-void statement_list_parse_treet::var_declarationt::output(
-  std::ostream &out) const
+void statement_list_parse_treet::networkt::add_instruction(instructiont &inst)
 {
-  out << variable.pretty() << "\n";
-  out << "  * initial_value: (default)";
+  instructions.push_back(inst);
 }
 
-void statement_list_parse_treet::networkt::output(std::ostream &out) const
+statement_list_parse_treet::tia_modulet::tia_modulet(
+  const irep_idt &name,
+  const float version)
+  : name(name), version(version)
 {
-  out << "Title: " << title.value_or("(none)") << "\n";
-  out << "Instructions: ";
-  if(instructions.empty())
-    out << "(none)";
-  out << "\n";
-  for(const auto &instruction : instructions)
-  {
-    instruction.output(out);
-    out << "\n";
-  }
 }
 
-void statement_list_parse_treet::instructiont::output(std::ostream &out) const
+statement_list_parse_treet::function_blockt::function_blockt(
+  const irep_idt &name,
+  const float version)
```

Comment:
If the version is something like `1.0`, `1.1`, etc., you're probably better off storing it in a `std::string`.